src/main/java/org/broadinstitute/hellbender/tools/exome/IntervalBackedExonDB.java

```diff
@@ -0,0 +1,361 @@
+*  7.3 Survival. The following provisions shall survive the expiration or termination of this Agreement: Articles 1, 3, 4, 5 and Sections 2.2, 2.3, 7.3, and 7.4.
+*  7.4 Notice. Any notices under this Agreement shall be in writing, shall specifically refer to this Agreement, and shall be sent by hand, recognized national overnight courier, confirmed facsimile transmission, confirmed electronic mail, or registered or certified mail, postage prepaid, return receipt requested.  All notices under this Agreement shall be deemed effective upon receipt.
+*  7.5 Amendment and Waiver; Entire Agreement. This Agreement may be amended, supplemented, or otherwise modified only by means of a written instrument signed by all parties. Any waiver of any rights or failure to act in a specific instance shall relate only to such instance and shall not be construed as an agreement to waive any rights or fail to act in any other instance, whether or not similar. This Agreement constitutes the entire agreement among the parties with respect to its subject matter and supersedes prior agreements or understandings between the parties relating to its subject matter.
+*  7.6 Binding Effect; Headings. This Agreement shall be binding upon and inure to the benefit of the parties and their respective permitted successors and assigns. All headings are for convenience only and shall not affect the meaning of any provision of this Agreement.
+*  7.7 Governing Law. This Agreement shall be construed, governed, interpreted and applied in accordance with the internal laws of the Commonwealth of Massachusetts, U.S.A., without regard to conflict of laws principles.
+*/
+package org.broadinstitute.hellbender.tools.exome;
+
+import org.broadinstitute.hellbender.utils.GenomeLoc;
+
+import java.util.*;
+
+/**
+ * Exon data-base based on a list of intervals.
+ *
+ * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
+ */
+public class IntervalBackedExonDB implements ExonDB<GenomeLoc> {
+
+    private final Map<String,GenomeLoc> intervalsByName;
+
+    private final ArrayList<GenomeLoc> sortedIntervals;
+
+    /**
+     * Creates a exon data-base give a sorted list of intervals.
+     *
+     * @param intervals the input interval list. Is assumed to be sorted
+     * @throws NullPointerException if {@code intervals} is {@code null}.
+     */
+    public IntervalBackedExonDB(final List<GenomeLoc> intervals) {
+
+        intervalsByName = new HashMap<>(intervals.size());
+        sortedIntervals = new ArrayList<>();
+        if (intervals.isEmpty())
+            return;
+        int nextIndex = 0;
```

Comment:
can you extract this code below to a new method? Constructors should be as simple as possible and this code does some work - at least by extracting a method it will have a clearer purpose